### PR TITLE
Improve docs on numeric literals precedence

### LINKED ISSUE #38181: Documentation error?
In the documentation to [Mathematical Operations and Elementary Functions: Operator Precedence and Associativity](https://docs.julialang.org/en/v1/manual/mathematical-operations/#Operator-Precedence-and-Associativity), there is the following sentence:

> Numeric literal coefficients, e.g. 2x, are treated as multiplications with higher precedence than any other binary operation, and also have higher precedence than ^.

But wouldn't this mean that

```
x = 3
2x^2
```

should evaluate to 36, and not 18? Getting 18 is probably what most people would expect from the expression without reading the docs, so: Is this a mistake in the documentation?

### PR DESCRIPTION
**Pull request by p-zubieta**
Fixes #38181

(the current wording was introduced in #32757)

### CODE REVIEW

File: doc/src/manual/mathematical-operations.md [hunk 1]
```diff
@@ -401,7 +401,15 @@ For a complete list of *every* Julia operator's precedence, see the top of this
 [`src/julia-parser.scm`](https://github.com/JuliaLang/julia/blob/master/src/julia-parser.scm). Note that some of the operators there are not defined
 in the `Base` module but may be given definitions by standard libraries, packages or user code.
 
-[Numeric literal coefficients](@ref man-numeric-literal-coefficients), e.g. `2x`, are treated as multiplications with higher precedence than any other binary operation, and also have higher precedence than `^`.
+[Numeric literal coefficients](@ref man-numeric-literal-coefficients), e.g. `2x`, are treated as multiplications with higher precedence than any other binary operation, with the exception of `^` where they have higher precedence only as the exponent.
+
+```jldoctest
+julia> let x = 3; 2x^2 end
```
Comment (fredrikekre) [inline doc/src/manual/mathematical-operations.md:407]:
```suggestion
julia> x = 3; 2x^2
```

File: doc/src/manual/mathematical-operations.md [hunk 2]
```diff
@@ -401,7 +401,15 @@ For a complete list of *every* Julia operator's precedence, see the top of this
...
+18
+
+julia> let x = 3; 2^2x end
```
Comment (fredrikekre) [inline doc/src/manual/mathematical-operations.md:410]:
```suggestion
julia> x = 3; 2^2x
```

File: doc/src/manual/mathematical-operations.md [hunk 3]
```diff
@@ -401,7 +401,15 @@ For a complete list of *every* Julia operator's precedence, see the top of this
...
-[Numeric literal coefficients](@ref man-numeric-literal-coefficients), e.g. `2x`, are treated as multiplications with higher precedence than any other binary operation, and also have higher precedence than `^`.
+[Numeric literal coefficients](@ref man-numeric-literal-coefficients), e.g. `2x`, are treated as multiplications with higher precedence than any other binary operation, with the exception of `^` where they have higher precedence only as the exponent.
+
+```jldoctest
+julia> x = 3; 2x^2
+18
+
+julia> x = 3; 2^2x
+64
+```
```
Comment (StefanKarpinski) [inline doc/src/manual/mathematical-operations.md:413]:
```suggestion

Juxtaposition parses like a unary operator, which has the same natural asymmetry around exponents: `-x^y` and `2x^y` parse as `-(x^y)` and `2(x^y)` whereas `x^-y` and `x^2y` parse as `x^(-y)` and `x^(2y)`.
```

**Comment by pabloferz** [comment]
Decided to move the juxtaposition bit down, because I feel it reads better. It was kind of in the middle of another explanation on operator precedence.

**Pull request merged**